watertap/unit_models/osmotically_assisted_reverse_osmosis_base.py

```diff
@@ -543,11 +543,8 @@ def initialize_build(
 
         init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
 
-        if not check_optimal_termination(res):
-            if raise_on_failure:
-                raise InitializationError(
-                    f"Unit model {self.name} failed to initialize"
-                )
+        if not check_optimal_termination(res) and raise_on_failure:
```

Comment:
I would like to get rid of `raise_on_failure` here, but we need IDAES/idaes-pse#1008 (or something like it).

Reply:
Yes, it's used in the LSRRO flowheet initialization here: https://github.com/watertap-org/watertap/blob/8fe2e562abd18dc7e16a951e6724039f67f1e911/watertap/examples/flowsheets/lsrro/lsrro.py#L716

In order to (straightforwardly) replace its usage with try/except I'd need the error to be properly handled on the IDAES side as well, hence https://github.com/IDAES/idaes-pse/pull/1008.

Keep in mind that the default for `raise_on_failure` is `False`, so the exception will be raised unless the user specifically request it not be.

Reply:
I don't think I am following here - for one, I can't actually see where `osmotically_assisted_reverse_osmosis_base.py` comes into that example.

However, the step to raise the `InitializationError` in that file is the last lines in the `initialize` method so I am not sure why whatever ends up calling that method would not be able to just wrap it in a `try/except`. As far as I can tell, all the steps to revert the model state have already been performed.